File: 2_AddRobot/ws/src/gpu_slam_bot/src/auto_move.cpp
#include <chrono>
#include <memory>
#include "rclcpp/rclcpp.hpp"
#include "geometry_msgs/msg/twist.hpp"

using namespace std::chrono_literals;

class AutoMove : public rclcpp::Node
{
public:
    AutoMove() : Node("auto_move")
    {
        pub_ = this->create_publisher<geometry_msgs::msg::Twist>("cmd_vel", 10);
        timer_ = this->create_wall_timer(
            100ms, std::bind(&AutoMove::timer_callback, this)
        );
    }

private:
    void timer_callback()
    {
        auto msg = geometry_msgs::msg::Twist();
        msg.linear.x = 0.2;   // forward speed
        msg.angular.z = 0.0;  // no rotation
        pub_->publish(msg);
    }

    rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr pub_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<AutoMove>());
    rclcpp::shutdown();
    return 0;
}

File: 2_AddRobot/ws/src/gpu_slam_bot/gpu_slam_bot/auto_move.py
import rclpy
from rclpy.node import Node
from geometry_msgs.msg import Twist

class AutoMove(Node):
    def __init__(self):
        super().__init__('auto_move')
        self.pub = self.create_publisher(Twist, 'cmd_vel', 10)
        timer_period = 0.1
        self.timer = self.create_timer(timer_period, self.timer_callback)

    def timer_callback(self):
        msg = Twist()
        msg.linear.x = 0.2  # forward speed
        msg.angular.z = 0.0 # no rotation
        self.pub.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = AutoMove()
    rclpy.spin(node)
    rclpy.shutdown()
